src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java

```diff
@@ -1609,6 +1609,13 @@ public ScrollListener(int dir, boolean block)   {
 
         /** {@inheritDoc} */
         public void actionPerformed(ActionEvent e) {
+            // If scrollbar isn't visible, stop the timer
+            if (!scrollbar.isShowing()) {
+                ((Timer)e.getSource()).stop();
```

Comment:
It seems there is duplication of these lines with l1626-1628 which can be placed in a helper method..